plugins/aws/sqs/sqsPublicAccess.js

```diff
@@ -0,0 +1,107 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'SQS Public Access',
+    category: 'SQS',
+    description: 'Ensures that SQS queues are not publically accessible',
+    more_info: 'SQS queues should be not be publically accessible to prevent unauthorized actions.',
+    recommended_action: 'Update the SQS queue policy to prevent public access.',
+    link: 'http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-creating-custom-policies.html',
+    apis: ['SQS:listQueues', 'SQS:getQueueAttributes', 'STS:getCallerIdentity'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        var acctRegion = helpers.defaultRegion(settings);
+        var accountId = helpers.addSource(cache, source,
+            ['sts', 'getCallerIdentity', acctRegion, 'data']);
+        var awsOrGov = helpers.defaultPartition(settings);
+
+        async.each(regions.sqs, function(region, rcb){
+            var listQueues = helpers.addSource(cache, source,
+                ['sqs', 'listQueues', region]);
+
+            if (!listQueues) return rcb();
+
+            if (listQueues.err) {
+                helpers.addResult(results, 3,
+                    `Unable to query for SQS queues: ${helpers.addError(listQueues)}`, region);
+                return rcb();
+            }
+
+            if (!listQueues.data || !listQueues.data.length) {
```

Comment:
move listQueues.data to the error check above

Reply:
In this particular case, if there are no SQS queues, the API returns nothing and listQueues is just an empty array. It is implemented same in other sqs plugins as well